examples/deadline/All-In-AWS-Infrastructure-Basic/python/package/lib/compute_tier.py

```diff
@@ -43,6 +48,22 @@ class ComputeTierProps(StackProps):
     # The bastion host to  allow connection to Worker nodes.
     bastion: Optional[BastionHostLinux] = None
 
+class UserDataProvider(InstanceUserDataProvider):
+    def pre_cloud_watch_agent(self, host) -> None:
+        host.user_data.add_commands("echo preCloudWatchAgent")
+    def pre_deadline_configuration(self, host) -> None:
+        host.user_data.add_commands("echo preDeadlineConfiguration")
+    def post_worker_launch(self, host) -> None:
+        host.user_data.add_commands("echo postWorkerLaunch")
+        test_script=Asset(host.node.scope, "SampleAsset",
```

Comment:
I'm wondering if it would be better to have the base class (`InstanceUserDataProvider`) be a sub-class of `Construct`. That way, it has its own scope in the construct tree and any child constructs (like these assets) can be added to its scope in the constructor and not have to create constructs in the `host.node.scope` like this. It would look like this:

```py
class UserDataProvider(InstanceUserDataProvider):
    def __init__(self, scope: Construct, id: str):
        super().__init__(scope, id);
        self.test_script = Asset(self, 'test_script', path=...)

    def post_worker_launch(self, host: IHost) -> None:
        self.test_script.grant_read(host)
        local_path = host.user_data.add_s3_download_command(
            bucket=test_script.bucket,
            bucket_key=test_script.s3_object_key
        )
        host.user_data.add_execute_file_command(file_path=local_path)
```

what do you think?

Reply:
Resp;ved